env/src/test/java/com/sap/cloud/security/config/k8s/DefaultServiceManagerServiceTest.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * SPDX-FileCopyrightText: 2018-2021 SAP SE or an SAP affiliate company and Cloud Security Client Java contributors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.sap.cloud.security.config.k8s;
+
+import com.sap.cloud.security.client.HttpClientTestFactory;
+import com.sap.cloud.security.config.ClientCredentials;
+import com.sap.cloud.security.config.OAuth2ServiceConfiguration;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.*;
+
+class DefaultServiceManagerServiceTest {
+	static DefaultServiceManagerService cut;
+	private static String servicePlans;
+	private static String serviceInstances;
+	static CloseableHttpClient httpClientMock = mock(CloseableHttpClient.class);
+
+	@BeforeAll
+	static void beforeAll() throws IOException {
+		servicePlans = IOUtils.resourceToString("/k8s/servicePlans.json", StandardCharsets.UTF_8);
+		serviceInstances = IOUtils.resourceToString("/k8s/serviceInstances.json", StandardCharsets.UTF_8);
+
+		OAuth2ServiceConfiguration oAuth2ServiceConfiguration = mock(OAuth2ServiceConfiguration.class);
+		when(oAuth2ServiceConfiguration.getUrl()).thenReturn(URI.create("https://auth.sap.com"));
+		when(oAuth2ServiceConfiguration.getProperty("sm_url")).thenReturn("https://service-manager.sap.com");
+		when(oAuth2ServiceConfiguration.getClientIdentity())
+				.thenReturn(new ClientCredentials("clientId", "clientSecret"));
+		cut = new DefaultServiceManagerService(oAuth2ServiceConfiguration, httpClientMock);
+	}
+
+	@Test
+	void getServicePlans() throws IOException {
+		CloseableHttpResponse response = HttpClientTestFactory
```

Comment:
This integration might not be required, you can install HttpClientTestFactory as described here:
https://github.com/SAP/cloud-security-xsuaa-integration/tree/main/token-client#new-warning-in-productive-environment-provide-well-configured-httpclientfactory-service